src/benchmark_api_internal.cc

```diff
@@ -1,5 +1,16 @@
 #include "benchmark_api_internal.h"
 
+namespace benchmark {
```

Comment:
did IWYU do this?

first, fold the two namespaces together.

Second, forward declarations aren't completely safe as they can hide a dependency. I'd rather take the hit on compilation time by including the relevant headers than have incomplete dependency trees.

Reply:
I don't disagree that forward declarations aren't always completely safe,
especially if they are banned, like e.g. for STL.

But this is deep within the library, and does not manifest any issues thus far.
Please can you explain what kind of danger do you foresee here?

Reply:
https://google.github.io/styleguide/cppguide.html#Forward_Declarations